pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1617,9 +1619,9 @@ public List<String> getSchemaNames() {
   public List<String> getSchemaNames(@Nullable String databaseName) {
     List<String> schemas = _propertyStore.getChildNames(
         PinotHelixPropertyStoreZnRecordProvider.forSchema(_propertyStore).getRelativePath(), AccessOption.PERSISTENT);
-    if (schemas != null) {
-        return schemas.stream().filter(schemaName -> isPartOfDatabase(schemaName, databaseName))
-            .collect(Collectors.toList());
+    if (databaseName != null) {
```

Comment:
This will return all schemas when `databaseName` is `null`, shouldn't the `null` database name be considered as the `default` database and only return schemas under the `default` database?

Reply:
Good call. We need to loop over all schemas